build/cpp/kvstore/src/kvstore.cc

```diff
@@ -1,6 +1,41 @@
 #include "kvstore.h"
 
-#include <iostream>
+// key value store constructor
+kvstore::KVStore::KVStore(const std::string &store) {
+  store_ = "data/" + store;
+  ReadFile();
+}
+
+// read key value store
+void kvstore::KVStore::ReadFile() {
+  std::ifstream input_file(store_, std::ifstream::in);
+  KVMap kvmap;
+  kvmap.ParseFromIstream(&input_file);
+  for (const auto &kvpair : kvmap.kvpairs()) {
+    std::vector<std::string> v;
+    for (const auto &value : kvpair.values()) {
+      v.push_back(value);
+    }
+    kvstore_map_[kvpair.key()] = v;
+  }
+}
+
+// write key value store
+void kvstore::KVStore::WriteFile() {
+  KVMap kvmap;
+  std::vector<KVPair> v;
+  for (std::unordered_map<std::string, std::vector<std::string>>::iterator it =
+           kvstore_map_.begin();
+       it != kvstore_map_.end(); ++it) {
+    KVPair kvpair;
+    kvpair.set_key(it->first);
+    *kvpair.mutable_values() = {it->second.begin(), it->second.end()};
+    v.push_back(kvpair);
+  }
+  *kvmap.mutable_kvpairs() = {v.begin(), v.end()};
+  std::ofstream output_file(store_, std::ofstream::out | std::ofstream::trunc);
```

Comment:
std::ofstream::out is not necessary in the mode argument

http://www.cplusplus.com/reference/fstream/ofstream/ofstream/

Reply:
std::ofstream::out is not necessary, and was removed.